src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -207,6 +259,25 @@ private static bool ValidTagStructure(string tagName, Block tag, RewritingContex
             return !invalidStructure;
         }
 
+        private static bool PartialTag(Block tagBlock)
+        {
+            // No need to validate the tag end because in order to be a tag block it must start with '<'.
+            var tagEnd = tagBlock.Children.Last() as Span;
```

Comment:
Can't do that here, it dives into all the children so if the last child is a block it finds his last child. In this case if the last child is a block then it's automatically a partial tag.